rasa/utils/train_utils.py

```diff
@@ -342,6 +368,94 @@ def override_defaults(
     return config
 
 
+def update_confidence_type(component_config: Dict[Text, Any]) -> Dict[Text, Any]:
+    """Set model confidence to cosine if margin loss is used.
+
+    Args:
+        component_config: model configuration
+
+    Returns:
+        updated model configuration
+    """
+    # TODO: Remove this once model_confidence is set to cosine by default.
```

Comment:
Shouldn't we still warn about invalid config in this case? We should maybe replace it with a `raise`  then.

Reply:
This not invalid config. I expect this to happen when users who had trained their assistant with margin loss in the past update to the new version and just re-train without changing anything in the config. Since the default value of `model_confidence` is set to `softmax`(as we believe most users use `cross_entropy` loss), this is an attempt to update the value of `model_confidence` for them automatically. If users follow the advice on the changelog, they will anyways change it to `cosine` themselves. ;)